lib/bolt/cli.rb

```diff
@@ -410,25 +420,32 @@ def load_task_data(name, modulepath)
       module_name, file_name = name.split('::', 2)
       file_name ||= 'init'
 
-      env = Puppet::Node::Environment.create('bolt', modulepath)
-      Puppet.override(environments: Puppet::Environments::Static.new(env)) do
-        data = Puppet::InfoService::TaskInformationService.task_data(
-          env.name, module_name, name
-        )
-
-        file = data[:files].find { |f| File.basename(f, '.*') == file_name }
-        if file.nil?
-          raise Bolt::CLIError, "Failed to load task file for '#{name}'"
-        end
+      begin
+        env = Puppet::Node::Environment.create('bolt', modulepath)
+        Puppet.override(environments: Puppet::Environments::Static.new(env)) do
+          data = Puppet::InfoService::TaskInformationService.task_data(
+            env.name, module_name, name
+          )
 
-        metadata =
-          if data[:metadata_file]
-            JSON.parse(File.read(data[:metadata_file]))
-          else
-            {}
+          file = data[:files].find { |f| File.basename(f, '.*') == file_name }
+          if file.nil?
```

Comment:
I'm not sure how this can happen with the TaskInformationService

Reply:
I think we were being overly defensive. We have a ticket to replace all of this with the PAL API for loading and validating task parameters, so will leave as is.